auth/src/error.rs

```diff
@@ -27,8 +27,14 @@ pub enum Error {
     IncompleteCheckoutSession,
     #[error("Interacting with stripe resulted in error: {0}.")]
     Stripe(#[from] StripeError),
-    #[error("Missing subscription ID from the checkout session.")]
+    #[error("Missing subscription ID.")]
     MissingSubscriptionId,
+    #[error("found more than one subscription items with the same metadata id: {0}")]
+    DuplicateSubscriptionItems(String),
+    #[error("found no subscription item with the given metadata id: {0}")]
+    MissingSubscriptionItem(String),
+    #[error("stripe subscription is canceled")]
```

Comment:
Should we use a capital letter at the start to align with the rest of the message ?

Reply:
I pondered changing all of these to all lowercase, since I think in general error messages should be, but it felt like I was adding quite a few unnecessary changes to an already large PR. I'll keep them consistent for now and we can clean it up later if anyone feels strongly enough. :smile: